dht/dhtcore/Janitor.c

```diff
@@ -264,12 +268,21 @@ static void peersResponseCallback(struct RouterModule_Promise* promise,
         String* addr = Address_toString(from, promise->alloc);
         Log_debug(janitor->logger, "Got peers response from [%s]", addr->bytes);
     }
+
     struct Address_List* addresses =
         ReplySerializer_parse(from, result, janitor->logger, true, promise->alloc);
 
     struct Node_Two* parent = NodeStore_nodeForAddr(janitor->nodeStore, from->ip6.bytes);
     if (!parent) { return; }
 
+    int64_t* moreNodesPtr = Dict_getInt(result, CJDHTConstants_MORE_NODES);
+    uint64_t moreNodes = moreNodesPtr ? *moreNodesPtr : 0;
+    if (moreNodes > 1) {
+        // FIXME(madafoo) malformed getPeers response may cause infinite loop here,
+        // might need threshold control.
+        getPeersMill(janitor, from, moreNodes);
```

Comment:
Ahh, you can't do this. The mills are only called inside of the main loop in order to limit the number of packets which are sent by janitor. With this patch there is a possibility that everyone replies, paths are long and (somehow) the main loop sends a lot of requests, this will create a flood. Basically the rule is "no sending traffic except inside of the janitor maintenance loop" (and then it's one packet per cycle)
If you need an exception to this, ask but you need to set an upper bound on how much traffic you could possibly be sending.